src/processor/FHIRProcessor.ts

```diff
@@ -7,52 +7,72 @@ import {
-  ExportableInvariant
-} from '../exportable';
+import { ExportableConfiguration } from '../exportable';
+import { ConfigurationExtractor } from '../extractor';
+import { Package } from './Package';
 
 export class FHIRProcessor {
   public readonly structureDefinitions: any[] = [];
+  public readonly codeSystems: any[] = [];
+  public readonly valueSets: any[] = [];
+  public readonly implementationGuides: any[] = [];
   public readonly fhir: fhirdefs.FHIRDefinitions;
 
   constructor(fhir: fhirdefs.FHIRDefinitions) {
     this.fhir = fhir;
   }
 
-  process(
-    inputPath: string,
-    existingInvariants: ExportableInvariant[] = []
-  ): (
-    | ExportableProfile
-    | ExportableExtension
-    | ExportableCodeSystem
-    | ExportableValueSet
-    | ExportableConfiguration
-    | ExportableInvariant
-  )[] {
+  register(inputPath: string): void {
     const rawContent = JSON.parse(fs.readFileSync(inputPath, 'utf-8'));
 
     if (rawContent['resourceType'] === 'StructureDefinition') {
       // Profiles and Extensions are both made from StructureDefinitions
       // Invariants may be contained within StructureDefinitions
       this.structureDefinitions.push(rawContent);
-      logger.debug(`Processing contents of ${inputPath} as StructureDefinition.`);
-      return StructureDefinitionProcessor.process(rawContent, this.fhir, existingInvariants);
+      logger.debug(`Registered contents of ${inputPath} as StructureDefinition.`);
     } else if (rawContent['resourceType'] === 'CodeSystem') {
-      logger.debug(`Processing contents of ${inputPath} as CodeSystem.`);
-      return [CodeSystemProcessor.process(rawContent)];
+      this.codeSystems.push(rawContent);
+      logger.debug(`Registered contents of ${inputPath} as CodeSystem.`);
     } else if (rawContent['resourceType'] === 'ValueSet') {
-      logger.debug(`Processing contents of ${inputPath} as ValueSet.`);
-      return [ValueSetProcessor.process(rawContent)];
+      this.valueSets.push(rawContent);
+      logger.debug(`Registered contents of ${inputPath} as ValueSet.`);
     } else if (rawContent['resourceType'] === 'ImplementationGuide') {
-      return [ConfigurationProcessor.process(rawContent)];
+      this.implementationGuides.push(rawContent);
+    }
+  }
+
+  process(resources: Package) {
+    let config: ExportableConfiguration;
+    if (this.implementationGuides.length > 0) {
+      config = ConfigurationProcessor.process(this.implementationGuides[0]);
     } else {
-      return [];
+      config = ConfigurationExtractor.process([...this.structureDefinitions, ...this.codeSystems]);
     }
+    resources.add(config);
+    this.structureDefinitions.forEach(sd => {
+      try {
+        StructureDefinitionProcessor.process(sd, this.fhir, resources.invariants).forEach(
+          resource => {
+            resources.add(resource);
+          }
+        );
+      } catch (ex) {
+        logger.error(`Could not process StructureDefinition: ${ex.message}`);
```

Comment:
Will `ex.message` contain the info needed to identify _which_ SD could not be processed?  I think it might not?  In which case we'd want to add some type of identifier to this error message.